src/Sentry/Protocol/SpanRecorder.cs

```diff
@@ -0,0 +1,40 @@
+using System.Collections.Generic;
+
+namespace Sentry.Protocol
+{
+    /// <summary>
+    /// Records spans that belong to the same transaction.
+    /// </summary>
+    public class SpanRecorder
+    {
+        private const int MaxSpans = 1000;
+
+        private readonly object _lock = new object();
+        private readonly List<Span> _spans = new List<Span>();
+
+        /// <summary>
+        /// Records a span.
+        /// </summary>
+        public void Add(Span span)
+        {
+            lock (_lock)
```

Comment:
Taking a lock to add any and all Spans can possibly create contention if we have a bunch of spans getting added to the same transaction.

Lets revisit this later to add a capped list that won't require exclusive access on all `Add`

Maybe just a `Interlocked.Increment` so we keep track of the count of items in the `_span` and using `ConcurrentBag` would be an improvement. But we discuss/measure things later, this will do for now.